tabbycat/tournaments/models.py

```diff
@@ -242,17 +242,17 @@ def current_rounds(self):
 
         # For something this complicated it's easier just to get the entire
         # round set from the database, and process it in Python.
-        rounds = self.round_set.filter(completed=False).order_by('seq').annotate(
-                Count('debate')).select_related('break_category')
+        rounds = getattr(self, 'current_round_set',
+            self.round_set.filter(completed=False).annotate(Count('debate')))
         current_elim_rounds = {}
         for r in rounds:
             if not r.is_break_round:
                 return [r]  # short-circuit everything else
             elif r.debate__count > 0:
-                current_elim_rounds.setdefault(r.break_category, r)
+                current_elim_rounds.setdefault(r.break_category_id, r)
         return [
-            current_elim_rounds.get(category)
-            for category in self.breakcategory_set.order_by('seq')
+            current_elim_rounds.get(category.pk)
+            for category in self.breakcategory_set.all()
             if category in current_elim_rounds
```

Comment:
You missed the `.pk` in line 256, so this function now always returns an empty list when it's not short-circuited.

On another note, the `.order_by('seq')` actually was important, but I think this probably should've been commented rather than left implicit. The returned list is intended to be in the order specified by the tab director using the `seq` fields in break category. As for the first `order_by` in line 245, that makes sure that if there are multiple break rounds in a category that aren't completed but have debates, it's the first one that is considered to be the current round, rather than whichever happens to come first in the database queryset order. While it's technically redundant because of the `Meta.ordering` options set in `BreakCategory` and `Round`, it's made explicit here so because it's critical to the function of the method, as opposed to a user interface nicety.